### ISSUE TITLE
Add `-u` flag (potentially repeatable) to disable filtering

### ISSUE THREAD

**Issue by xmo-odoo** [semgrep/semgrep#4539]
Copy of https://github.com/returntocorp/semgrep/issues/1964#issuecomment-1011849002 on the recommendation of @ievans.

semgrep has support for `.gitignore` files and its own `.semgrepignore` (possibly `.ignore` as well? Unsure).

It does have a flag to skip `.gititgnore` (i don't think it has one for `.semgrepignore` as yet? but `--no-git-ignore` can be a bit of a ~~mouthful~~ fingerful, and it's unclear what other hiding behaviour semgrep will have in the future (e.g. hidden files).

Here `ripgrep` has individual flags for ignore sources (e.g. `--no-hidden`, `--binary`, and a slew of `--no-ignore-$X` related to various categories of ignore files).

But it also has a repeatable `--unrestricted` (`-u`) flag which allows bypassing those ignore categories conveniently:

> Reduce the level of "smart" searching. A single -u won't respect .gitignore (etc.) files (--no-ignore). Two -u flags will additionally search hidden files and directories (-./--hidden). Three -u flags will additionally search binary files (--binary).
>
> 'rg -uuu' is roughly equivalent to 'grep -r'.

This is extremely convenient at the command-line (using the tool interactively) when trying to maximise the search area e.g, you don't find what you expected, you can just pop in a `-u` or 3, obviously at a cost (since it's probably going to trawl through a lot more things).

I think semgrep could benefit from this.

**Comment by aryx** [comment]
@mmcqd recently added support for .semgrepignore in semgrep itself (it was only in the agent before). Maybe he has some feedback on this feature request.

**Comment by xmo-odoo** [comment]
> @mmcqd recently added support for .semgrepignore in semgrep itself (it was only in the agent before). Maybe he has some feedback on this feature request.

Makes sense, this feature suggestion is, in a way, a followup to that change.

**Comment by xmo-odoo** [comment]
![animal 3](https://user-images.githubusercontent.com/7571158/152739463-9ecc8d56-137f-4d26-a480-618be9425d14.png)

**Comment by mjambon** [comment]
To ignore `.semgrepignore` files, use `--x-ignore-semgrepignore-files` or `--x-semgrepignore-filename` (subject to change).

This is part of our [Semgrepignore v2](https://semgrep.dev/docs/semgrepignore-v2-reference) implementation that is the default behavior since Semgrep 1.117.0. It works like Gitignore.
